cinema/views.py

```diff
@@ -48,6 +81,29 @@ class MovieSessionViewSet(viewsets.ModelViewSet):
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = MovieSession.objects.all()
+        date = self.request.query_params.get("date")
+        movie_id = self.request.query_params.get("movie")
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+        if movie_id:
+            queryset = queryset.filter(movie__id=int(movie_id))
+        if self.action == "list":
+            queryset = (
+                queryset.select_related("movie", "cinema_hall")
+                .annotate(
+                    tickets_available=F(
+                        "cinema_hall__rows"
+                    ) * F(
+                        "cinema_hall__seats_in_row"
+                    ) - Count(
+                        "tickets"
+                    )
```

Comment:
The `tickets_available` annotation calculates the available tickets by subtracting the count of tickets from the total capacity. Ensure that this logic correctly reflects the business rules for ticket availability, especially if there are any additional constraints or conditions that need to be considered.